manifests/user.pp

```diff
@@ -32,28 +29,25 @@
 ) {
   include ::st2::params
 
-  $_robots_group_name = $st2::params::robots_group_name
   $_packs_group_name = $st2::params::packs_group_name
-
-  ensure_resource('group', $_robots_group_name, {
-    'ensure' => present,
-  })
+  $_ssh_dir = "/home/${name}/.ssh"
 
   ensure_resource('group', $_packs_group_name, {
     'ensure' => present,
   })
 
   if $create_sudo_entry {
-    ensure_resource('sudo::conf', $_robots_group_name, {
+    ensure_resource('sudo::conf', $name, {
       'priority' => '10',
-      'content'  => "%${_robots_group_name} ALL=(ALL) NOPASSWD: SETENV: ALL",
+      # note: passes in $name variable into template
+      'content'  => template('st2/etc/sudoers.d/user.erb'),
     })
```

Comment:
Under ubuntu xenial Vagrant, found that something breaks paswordless sudo for the default user after running the provisioner.
Will check later what's going on there.